src/Orleans.Core/Configuration/OptionLogger/IOptionsLogger.cs

```diff
@@ -0,0 +1,84 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace Orleans
+{ 
+    internal class ClientOptionsLogger : OptionsLogger, ILifecycleParticipant<IClusterClientLifecycle>
+    {
+        private int ClientOptionLoggerLifeCycleRing = 1000;
```

Comment:
this depends on how you want lifecycle ring to be organized. Sure we can have random int lying around the code base and no ones knows how many rings are there and what their order are. 

or we can have something like the silo lifecycle ring has, a class which holds the major rings. And then if we go with silo lifecycle pattern, then each ring has a meaning which implies what kind of services should be registered here. Then the queston is what is int.Minvalue ring? Anything which should run before silo start ring?

Reply:
As the system currently exists, the number of rings is the range of int.  We simply have marker rings to help users organize around.  Int.Min will always be the first ring (assuming anything is registered there), and Int.Max will always be the last,  There is definitely room for us to form more conventions around this, but if we want to ensure this logic is run first (at lease as 'first' as possible, we should register this at a ring that is before what we expect others to use.

Reply:
if we don't have a summary of all the rings used in our code base, then if a new participant is trying to register to the lifecycle, it will have difficulty to know which int it can pick. Without the relative knowledge of all the other rings are, picking a ring to join can be hard.